comms/storage/contentaccess/content_access_checker_test.go

```diff
@@ -0,0 +1,152 @@
+package contentaccess
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"encoding/json"
+	"net/url"
+	"testing"
+	"time"
+
+	"comms.audius.co/shared/utils"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestRecoverWallet(t *testing.T) {
+	signatureData := SignatureData{
+		Cid:         "Qmblah",
+		ShouldCache: true,
+		Timestamp:   time.Now().Unix(),
+		TrackId:     12345,
+	}
+
+	privKey, err := crypto.GenerateKey()
+	if err != nil {
+		t.Fatal("Key generation failed")
+	}
+	signature, err := utils.GenerateSignature(signatureData.toMap(), privKey)
+	if err != nil {
+		t.Fatalf("Failed to generate signature for %+v", signatureData)
+	}
+	expectedSigner := crypto.FromECDSAPub(&privKey.PublicKey)
+
+	actualSigner, err := recoverWallet(signatureData, signature)
+	if err != nil {
+		t.Fatalf("recover wallet errored with %+v", err)
+	}
+
+	if !bytes.Equal(actualSigner, expectedSigner) {
+		t.Fatalf("public key doesn't match signer, want=%+v, got=%+v", expectedSigner, []byte(actualSigner))
+	}
+}
+
+func TestIsExpired(t *testing.T) {
+	tests := []struct {
+		data    SignatureData
+		expired bool
+	}{
+		{SignatureData{Timestamp: time.Now().Unix()}, false},
+		{SignatureData{Timestamp: 0}, true},
+	}
+
+	for _, tt := range tests {
+		actual := isExpired(tt.data)
+
+		if actual != tt.expired {
+			t.Fatalf("is expired failed for %+v, want=%+v, got=%+v", tt.data, tt.expired, actual)
+		}
+	}
+}
+
+func TestIsCidMatch(t *testing.T) {
+	tests := []struct {
+		data     SignatureData
+		cid      string
+		expected bool
+	}{
+		{SignatureData{Cid: "Qmblah"}, "Qmblah", true},
+		{SignatureData{Cid: "Qmblah"}, "blah", false},
+		{SignatureData{Cid: ""}, "blah", false},
+		{SignatureData{Cid: "Qmblah"}, "", false},
+	}
+
+	for _, tt := range tests {
+		actual := isCidMatch(tt.data, tt.cid)
+
+		if actual != tt.expected {
+			t.Fatalf("cid match failed for %+v, want=%+v, got=%+v", tt.cid, tt.expected, actual)
+		}
+	}
+}
+
+func TestVerifySignature(t *testing.T) {
+
```

Comment:
@theoilie is helping me write this test to make sure discovery nodes get mocked properly

Reply:
hmmm. I'd have to also make `VerifySignature()` accept `config.ServiceNode` too to properly test it which isn't really what I was going for :/ 

I can do it though since it seems that's the quickest way to get this PR merged